sympy/stats/frv_types.py

```diff
@@ -174,14 +175,10 @@ def Die(name, sides=6):
 class BernoulliDistribution(SingleFiniteDistribution):
     _argnames = ('p', 'succ', 'fail')
 
-    def __new__(cls, *args):
-        p = args[BernoulliDistribution._argnames.index('p')]
-        p_sym = sympify(p)
-
-        if fuzzy_not(fuzzy_and((p_sym.is_nonnegative, (p_sym - 1).is_nonpositive))):
-            raise ValueError("p = %s is not in range [0, 1]." % str(p))
-        else:
-            return super(BernoulliDistribution, cls).__new__(cls, *args)
+    @staticmethod
+    def check(p, succ, fail):
+        _value_check((p >= 0) != False and (p <= 1) != False,
```

Comment:
->`And(p >= 0, p <= 1)`

Reply:
I have removed `And` because of the [this](https://github.com/sympy/sympy/pull/16576#discussion_r282803523) discussion. What are your views?